Update the Terms of Service to make it more clearer (#38978)
is-a-dev/register: TERMS_OF_SERVICE.md

For questions or concerns about these Terms, please contact us at admin@is-a.dev.

> **Last Updated**: 23th May, 2026

Reviewer: ```suggestion
**Last Updated**: 23rd May, 2026
```

> **Last Updated**: 2nd June, 2026

Reviewer 2: ```suggestion
**Last Updated**: 4th June, 2026
```

## Definitions

> Before you continue reading the Terms of Service, here are the following definitions that apply within the Terms of Service:

Reviewer: ```suggestion
The following definitions apply within these Terms of Service:
```

- **"Service"** refers to the is-a.dev subdomain registration service, website, infrastructure, GitHub repository, and related systems operated by is-a.dev.
- **"User"**, **"you"**, and **"your"** refer to any individual using or accessing the service.
- **"Subdomain"** refers to any subdomain registered under the `is-a.dev` domain.
- **"Root subdomain"** refers to the primary registered subdomain itself, excluding any nested subdomains. For example, `example.is-a.dev` is a root subdomain, while `blog.example.is-a.dev` is a **"nested subdomain"**.

> - **"Content"** refers to any website, files, media, software, text, or other material hosted or served using a subdomain.

Reviewer: ```suggestion
- **"Content"** refers to any website, files, media, software, text, or other material hosted or served using on an is-a.dev subdomain.
```

## 1. Acceptance of Terms

> By accessing or using is-a.dev, you acknowledge that you have read, understood, and agree to be bound by these Terms. If you do not agree, you cannot use the is-a.dev service.

Reviewer: ```suggestion
By accessing or using is-a.dev, you acknowledge that you have read, understood, and agree to be bound by these Terms. If you do not agree, you may not use the is-a.dev service.
```
there was no need to change may not to cannot

## 2. Eligibility
To use is-a.dev, you must:

1. Be at least 13 years old;

> 2. Be an individual person, not an organisation, acting either on your own behalf or as a representative of a group or a non-commercial project;

Reviewer: ```suggestion
2. Be an individual person, not an organisation, acting either on your own behalf or as a representative of a group or non-commercial project;
```

3. Use the service for lawful purposes only;
4. Provide accurate and truthful information when registering.

> Subdomains must be registered to an individual GitHub account, organisations cannot be the registered owner of a subdomain.

Reviewer: ```suggestion
Subdomains must be registered to an individual GitHub account, and cannot be registered to a GitHub organisation account.
```

> Subdomains must be registered to an individual GitHub account, organisations should not be the registered owner of a subdomain.

Reviewer 3: "cannot" instead of "should not" might be clearer and more consistent.

By using is-a.dev, you represent and warrant that you meet these requirements.

## 3. Service Description
is-a.dev allows users to register a subdomain under the "is-a.dev" domain. This service is offered "as-is" without guarantees of uptime, support, or permanence.

> Subdomains are intended solely for individuals, software developer groups and non-commercial projects that are related to software developement, as long as it doesn't contain activities that are prohibited.

Reviewer 3: ```suggestion
Subdomains are intended solely for individuals, software developer groups and non-commercial projects that are related to software development, as long as the project does not contain activities that are prohibited by us.
```

Reviewer 4: It might be worth changing `prohibited by us` to `prohibited under these Terms`.

`Prohibited by us` reads as though the prohibition comes from the authors or maintainers personally, whereas `prohibited under these Terms` makes it clear that the restriction originates from the Terms themselves. It's also the more common wording used in legal and policy documents.

Edit: Seen it's been changed, dw lolll

> Subdomains are intended solely for individuals, software developer groups and non-commercial projects that are related to software development, as long as the project does not contain activities that are prohibited by us.

Reviewer: ```suggestion
Subdomains are intended solely for individuals, software developer groups and non-commercial projects that are related to software development, as long as the project does engage in activities or serve content that is prohibited by these Terms of Service.
```

For questions or concerns about these Terms, please contact us at admin@is-a.dev.

**Last Updated**: 23th May, 2026

## Definitions
Before you continue reading the Terms of Service, here are the following definitions that apply within the Terms of Service:
- **"Service"** refers to the is-a.dev subdomain registration service, website, infrastructure, GitHub repository, and related systems operated by is-a.dev.
- **"User"**, **"you"**, and **"your"** refer to any individual using or accessing the service.
- **"Subdomain"** refers to any subdomain registered under the `is-a.dev` domain.
- **"Root subdomain"** refers to the primary registered subdomain itself, excluding any nested subdomains. For example, `example.is-a.dev` is a root subdomain, while `blog.example.is-a.dev` is a **"nested subdomain"**.
- **"Content"** refers to any website, files, media, software, text, or other material hosted or served using a subdomain.

## 1. Acceptance of Terms

> By accessing or using is-a.dev, you acknowledge that you have read, understood, and agree to be bound by these Terms. If you do not agree, you may not use is-a.dev.

Reviewer 5: ```suggestion
By accessing or using is-a.dev, you acknowledge that you have read, understood, and agree to be bound by these Terms. If you do not agree, you cannot use the is-a.dev service.
```

Author: Isn't `you cannot not use the is-a.dev service` better?

Reviewer 5: No, it's double negetive. `cannot`, `not`

Reviewer 5: This is better, `If you do not agree, you must not access or use the is-a.dev service.`, let me edit the suggestion. 
It covers both type, those who want to use, and those who are already using.

Author: > No, it's double negetive. `cannot`, `not`

I made a typo, oops. It was supposed to not have 'not'.

> This is better, `If you do not agree, you must not access or use the is-a.dev service.`, let me edit the suggestion.

I'm leaning towards `you cannot use the is-a.dev service`, I don't think "you must not access" is fine because accessing means going to a is-a.dev domain for example.

Reviewer 5: > "you must not access" is fine because accessing means going to a is-a.dev domain for example.

Yeah, i missed that thing. Okay i fixed that.

14. Blogs that are not primarily related to software development
15. Conversational AI websites or apps that are similar in functionality to websites or apps such as Claude, Gemini, ChatGPT, etc.

*Note: This list is not exhaustive and may be updated or expanded at any time at our discretion.*

Additionally, users must comply with [GitHub's Terms of Service](https://docs.github.com/en/site-policy/github-terms/github-terms-of-service) where applicable.

Violation of this section may result in immediate termination without notice.

## 5. Use of AI (Artificial Intelligence)

The use of AI for creating, designing, or modifying websites is acceptable as long as the website is complete and complies with the prohibited activities guidelines. We are strongly against the usage of AI (such as Claude Code, Codex, OpenClaw, etc) for creating pull requests, and may close the pull request alongside blocking or limiting the author's ability to interact with our repositories and resources, if used.

## 6. Ownership and Intellectual Property
is-a.dev retains all rights, title, and interest in the "is-a.dev" domain and associated systems.

> You retain ownership of any content you host on your subdomain. Your hosted content, however, must not infringe third-party rights.

Reviewer 2: ```suggestion
You retain ownership of any content you host on your subdomain. Your hosted content, however, must not infringe on third-party rights.
```

6. Impersonation and misrepresentation

> 7. Infringe third-party rights

Reviewer: ```suggestion
7. Infringing third-party rights
```
we must keep the tense consistent.

> 8. Commercial, for-profit or political purposes

Reviewer: ```suggestion
8. Commercial, for-profit, or political purposes
```
oxford comma smh my head

> 9. Sending unsolicited communications (spam)

Reviewer 3: ```suggestion
9. Sending unsolicited communications ("spam")
```
This format is more commonly used, as I've seen in legal documents.

> 9. Sending unsolicited communications ("spam")

Reviewer: ```suggestion
9. Sending unsolicited communications or "spam"
```

10. Cryptocurrency and blockchain mining

> 11. Abusive automation (e.g. bots, scrapers, or scripts that place excessive load on the service)

Reviewer 3: ```suggestion
11. Abusive automation (including but not limited to: "bots", "scrapers", or automated scripts that place excessive load on the service)
```

12. Proxy or VPN services

> 13. Websites on root subdomains that are behind a login page (login page for management are allowed)

Reviewer 3: ```suggestion
13. Websites on root subdomains that are behind a login page (with the exception of login pages for management reasons)
```
Grammatical error as well, along with being somewhat ambiguous.

> 13. Websites on root subdomains that are behind a login page (with the exception of login pages for management reasons)

Reviewer: ```suggestion
13. Hosting websites root subdomains that are behind a login page (with the exception of login pages for management reasons)
```

14. Blogs that are not primarily related to software development

> 15. AI websites that mimick sites like Claude, Gemini, ChatGPT and etc.

Reviewer 3: ```suggestion
15. Conversational AI websites or apps that are similar in functionality to websites or apps such as Claude, Gemini, ChatGPT, etc.
```
Many grammatical errors, and some ambiguity.

> *Note: This list is not exhaustive and may be updated or expanded at any time at our discretion. We also reserve the right to deny, reject, or revoke any subdomain registration or pull request for reasons not explicitly listed here, including but not limited to reasons identified during the review process.*

Reviewer 3: This creates a duplicate of line 36 ("We reserve the right to deny any subdomain registration request for any reason.").

Additionally, users must comply with [GitHub's Terms of Service](https://docs.github.com/en/site-policy/github-terms/github-terms-of-service) where applicable.

Violation of this section may result in immediate termination without notice.

## 5. Use of AI (Artificial Intelligence)

> The use of AI for creating, designing or modifying websites is acceptable as long as the website is complete and abides by the prohibited activities guidelines. We are strongly against using AI (such as Claude Code, Codex, OpenClaw, etc) for creating pull requests, and may close the pull request alongside blocking or limiting the author's ability to interact with our repositories and resources, if used.

Reviewer: ```suggestion
The use of AI for creating, designing, or modifying websites is acceptable as long as the website is complete and complies with the prohibited activities guidelines. We are strongly against the usage of AI (such as Claude Code, Codex, OpenClaw, etc) for creating pull requests, and may close the pull request alongside blocking or limiting the author's ability to interact with our repositories and resources, if used.
```

> The use of AI for websites is acceptable as long as the website is complete. For pull requests that are generated with AI (e.g. OpenClaw, Claude Code, Codex) we are strongly against and can lead to the pull request being closed alongside blocking or limiting the author's ability to interact with is-a.dev's repositories and resources.

Reviewer 3: ```suggestion
The use of AI for creating, designing or modifying websites is acceptable as long as the website is complete and abides by the prohibited activities guidelines. We are strongly against using AI (such as Claude Code, Codex, OpenClaw, etc) for creating pull requests, and may close the pull request alongside blocking or limiting the author's ability to interact with our repositories and resources, if used.
```
This one is really hard to read, and has ambiguity.

## 6. Ownership and Intellectual Property
is-a.dev retains all rights, title, and interest in the "is-a.dev" domain and associated systems.

> You retain ownership of any content you host on your subdomain. Your hosted content however must not infringe third-party rights.

Reviewer: ```suggestion
You retain ownership of any content you host on your subdomain. Your hosted content, however, must not infringe third-party rights.
```

You grant us permission to use your registration data to operate the service.

## 7. Termination of Service

> We reserve the right to terminate your subdomain at any time, for any reason. While we will make reasonable efforts to provide notice in advance, termination may occur without notice in cases of serious or urgent violations.

Reviewer 5: ```suggestion
We reserve the right to terminate your subdomain at any time, for any reason. While we make reasonable efforts to provide notice in advance, we are not obligated to do so.
```

Author: Wouldn't this be better? `While we try to make reasonable efforts to provide advance notice, we are not obligated to do so`

Reviewer 5: no, `we try to` is weaker than `we make` in legal terms. And `we make` is standard.

Author: Saw I made a typo and you added that change, so I added it for you.

Any donations made to the service are non-refundable.

## 8. Privacy
We collect and process only the personal data strictly necessary to operate is-a.dev. We do not sell or share your data with third parties for marketing purposes. While we take reasonable precautions, we are not responsible for any data or privacy breaches resulting from the use of our service.

For privacy-related inquiries, please contact us at admin@is-a.dev.

## 9. Disclaimer and Limitation of Liability
The is-a.dev service is provided "as-is" and "as available." We make no warranties, expressed or implied, regarding the service's functionality, reliability, or suitability for any particular purpose.

You use the service at your own risk.

## 10. Indemnification
You agree to indemnify and hold harmless is-a.dev, its operators, and affiliates from any claims, losses, liabilities, or expenses arising from your use of the service or violation of these Terms.

## 11. Governing Law and Jurisdiction

> These Terms are governed by the laws of Australia. Any disputes will be subject to the exclusive jurisdiction of Australian courts.

Reviewer 5: ```suggestion
These Terms are governed by the laws of Australia. Any disputes arising out of or relating to these Terms will be subject to the exclusive jurisdiction of the courts of Australia.
```

Author: Not 100% sure if this is necessary.

Reviewer 5: Actually `Arising out of or relating to` is standard form, simply saying `any disputes` can sometimes be interpreted narrowly and it is also too broad not legal type.
